lib/config/index.js

```diff
@@ -136,4 +138,30 @@ function checkLimitsConfig (cfg) {
   }
 }
 
+/** Check the format of the error-codes config files
+ *
+ */
+
+function checkErrorCodes() {
+
+  for (const domain of Object.keys(errorCodes)) {
+    let name = errorCodes[domain];
+    if (!name.hasOwnProperty('subdomains') || !name.hasOwnProperty('code')) {
+      throw new KuzzleInternalError(`Invalid configuration: Missing mandatory field in ${domain} error-codes config file.`, name, 0);
+    }
+    for (const subdomain of Object.keys(errorCodes[domain]['subdomains'])) {
+      name = errorCodes[domain]['subdomains'][subdomain];
```

Comment:
`name` is overwritten multiple times, each time with a different kind of value (domain, subdomain, ...)

use different constants instead

Reply:
SonarQube forbids me to do that, if the `const` have the same name. If I use 3 const, I must name them differently, I don't see how it's accurate
What is the problem of overwriting a single variable ?

Reply:
> SonarQube forbids me to do that if the 3 const have the same name.

And that's a good thing. But nothing prevents you from having a const named `domain`, another one named `subdomain`, and so on.

The issue is not with the overwriting itself, but with the different meanings you give to it: depending on where we are in the function, `name` can contain something entirely different than 2 lines above. This is a bad practice because it increases the cognitive complexity of your function.

Reply:
More information: this is usually detected by Sonarqube under certain conditions, with the following rule: https://rules.sonarsource.com/javascript/type/Code%20Smell/RSPEC-1524